tests/Unit/Adapter/ValidateTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Validate;
+
+class ValidateTest extends TestCase
+{
+    /** @var Validate */
+    private $validate;
+
+    public function setUp()
+    {
+        $this->validate = new Validate();
+    }
+
+    /**
+     * @dataProvider getUrls
+     *
+     * @param string $url
+     * @param bool $expected
+     */
+    public function testValidateIsUrl(string $url, bool $expected)
+    {
+        $this->assertEquals($expected, $this->validate->isURL($url));
+    }
+
+    public function getUrls()
+    {
+        return [
+            ['https://prestashop.com', true],
+            ['http://prestashop.com/', true],
+            ['https://prestashop.com/demo', true],
+            ['http://presat+shop.com/demo', true],
```

Comment:
It's not a valid URL. + in the domain is forbidden :thinking: 
A complete domain name is valid if it meets the following criteria:

A complete domain name must have one or more subdomain names and one top-level domain name.
A complete domain name must use dots (.) to separate domain names.
Domain names must use only alphanumeric characters and dashes (-).
Domain names must not begin or end with dashes (-).
Domain names must mot have more than 63 characters.
The top-level domain name must be one of the predefined top-level domain names, like (com), (org), or (ca)

Reply:
@PierreRambaud I totally agree. The problem is that I'm testing a legacy function that I don't think I should change if we don't want to have BC Breaks. That being said, instead of using that legacy function, I can use a new one that really check urls.

Reply:
You shouldn't consider the function as a BC break if you fix a bug. In URL you're able to add + char, the method is for the moment invalid.
If you allow '+' without checking the domain you're introducing a bug :thinking:

Reply:
Be careful, there is 2 methods here. The one where I allowed the "+" sign is not the legacy function being tested. The other one being tested was already allowing the "+" sign. That's why I thought I should create a new method to replace the legacy one.

Reply:
I mean, the name of the function is still `isUrl` or `isValidUrl`, and by allowing `+` in the domain it will return a wrong value.
This is why it's confusing me :sweat_smile: